### PR TITLE
OPENEUROPA-2092: ECL 2.0: Link blocks

### PR DESCRIPTION
**Pull request by voidtek**
### Description

Add link-block template and its tests.

### CODE REVIEW

File: templates/compositions/ec-component-link-block/link-block.html.twig [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{% spaceless %}
+
+{#
+  /**
+   * @file
+   * Theme implementation for a ECL Link Block.
+   */
+    Parameters:
+      - "title" (string) (default: ''): optional title of the link block
+      - "links" (array) (default: []): collection of standalone @ecl/generic-component-link
```
Comment (sergepavle) [inline templates/compositions/ec-component-link-block/link-block.html.twig]: Do we still have this generic-component-link template in ECL2 ?

File: templates/compositions/ec-component-link-block/link-block.html.twig [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      <h5 class="ecl-u-type-heading-5 ecl-u-mt-xs ecl-u-mb-none">{{ _title }}</h5>
+    </div>
+  {% endif %}
+  {% if _links is not empty and _links is iterable %}
+    <div class="ecl-row ecl-u-pl-m ecl-u-pr-m">
+      <ul class="ecl-link-block__list ecl-unordered-list--no-bullet ecl-u-ma-none">
+        {% for _link in _links %}
+          <li class="ecl-link-block__item ecl-u-mt-xs ecl-u-mb-xs">
+            {% include '@ecl/link' with {
+              extra_classes: 'ecl-link-block__link'
```
[the remark this answers is not in the document]
Comment (voidtek) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: This class doesn't contain any CSS attributes, but very helpful for selectors.
Actually, we add this class on the composition only to help ECL-twig team for future implementations, adding a new class for the component.

File: templates/compositions/ec-component-link-block/link-block.html.twig [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  {% if _title is not empty %}
+    <div class="ecl-link-block__title ecl-row ecl-u-pl-m ecl-u-pr-m">
+      <h5 class="ecl-u-type-heading-5 ecl-u-mt-xs ecl-u-mb-none">{{ _title }}</h5>
+    </div>
+  {% endif %}
+  {% if _links is not empty and _links is iterable %}
+    <div class="ecl-row ecl-u-pl-m ecl-u-pr-m">
+      <ul class="ecl-link-block__list ecl-unordered-list--no-bullet ecl-u-ma-none">
+        {% for _link in _links %}
+          <li class="ecl-link-block__item ecl-u-mt-xs ecl-u-mb-xs">
```
[the remark this answers is not in the document]
Comment (voidtek) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: This class doesn't contain any CSS attributes, but very helpful for selectors.
Actually, we add this class on the composition only to help ECL-twig team for future implementations, adding a new class for the component.

File: templates/compositions/ec-component-link-block/link-block.html.twig [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+<div class="{{ _css_class }} ecl-container ecl-u-bg-grey-5"{{ _extra_attributes|raw }}>
+  {% if _title is not empty %}
+    <div class="ecl-link-block__title ecl-row ecl-u-pl-m ecl-u-pr-m">
+      <h5 class="ecl-u-type-heading-5 ecl-u-mt-xs ecl-u-mb-none">{{ _title }}</h5>
+    </div>
+  {% endif %}
+  {% if _links is not empty and _links is iterable %}
+    <div class="ecl-row ecl-u-pl-m ecl-u-pr-m">
+      <ul class="ecl-link-block__list ecl-unordered-list--no-bullet ecl-u-ma-none">
```
[the remark this answers is not in the document]
Comment (voidtek) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: This class doesn't contain any CSS attributes, but very helpful for selectors.
Actually, we add this class on the composition only to help ECL-twig team for future implementations, adding a new class for the component.

File: templates/compositions/ec-component-link-block/link-block.html.twig [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+<div class="{{ _css_class }} ecl-container ecl-u-bg-grey-5"{{ _extra_attributes|raw }}>
+  {% if _title is not empty %}
+    <div class="ecl-link-block__title ecl-row ecl-u-pl-m ecl-u-pr-m">
```
[the remark this answers is not in the document]
Comment (voidtek) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: This class doesn't contain any CSS attributes, but very helpful for selectors.
Actually, we add this class on the composition only to help ECL-twig team for future implementations, adding a new class for the component.

File: templates/compositions/ec-component-link-block/link-block.html.twig [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      - "system" (string) (default: "generic")
+#}
+
+{# Internal properties #}
+
+{% set _title = title|default('') %}
+{% set _links = links|default([]) %}
+{% set _css_class = 'ecl-link-block' %}
+{% set _extra_attributes = '' %}
+{% set _system = system|default('generic') %}
```
[the remark this answers is not in the document]
Comment (voidtek) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: Good catch.

File: templates/compositions/ec-component-link-block/link-block.html.twig [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      - "links" (array) (default: []): collection of standalone @ecl/generic-component-link
+      - "extra_classes" (string) (default: '')
+      - "extra_attributes" (array) (default: []): format: [
+          {
+            "name" (string) (default: ''),
+            "value" (string) (default: '')
+          },
+          ...
+        ]
+      - "system" (string) (default: "generic")
```
[the remark this answers is not in the document]
Comment (voidtek) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: good catch.

File: templates/compositions/ec-component-link-block/link-block.html.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (voidtek) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: Good catch.

**Comment by voidtek** [comment]
I rebased the PR, comment some remarks and push changes.

File: templates/compositions/ec-component-link-block/link-block.html.twig (hunk 2 above)
[the remark this answers is not in the document]
Comment (sergepavle) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: If it is ok for ECL-twig team, ok for me

File: templates/compositions/ec-component-link-block/link-block.html.twig (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergepavle) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: If it is ok for ECL-twig team, ok for me

File: templates/compositions/ec-component-link-block/link-block.html.twig (hunk 4 above)
[the remark this answers is not in the document]
Comment (sergepavle) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: If it is ok for ECL-twig team, ok for me

File: templates/compositions/ec-component-link-block/link-block.html.twig (hunk 5 above)
[the remark this answers is not in the document]
Comment (sergepavle) [reply templates/compositions/ec-component-link-block/link-block.html.twig]: If it is ok for ECL-twig team, ok for me

**Pull request merged**